### PR TITLE
Fixy fix #3644.

### LINKED ISSUE #3644: Assasinate clown objective still lists random clown name instead of clown true name
Seen on a vampire round. Clown was targeted as Bo Bo Sassy when clown was not named as such. :hankey:

### PR DESCRIPTION
**Pull request by eswordthecat**

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -174,59 +174,74 @@ Turf and target are seperate in case you want to teleport some distance from a t
...
+				var/obj/item/device/pda/PDA = object
+
+				if (PDA.owner == oldname)
 					PDA.owner = newname
 					PDA.name = "PDA-[newname] ([PDA.ownjob])"
-					if(!search_id)	break
-					search_pda = 0
-
-		//Fixes renames not being reflected in objective text
-		var/list/O = (typesof(/datum/objective)  - /datum/objective)
```
Comment (eswordthecat) [inline code/__HELPERS/unsorted.dm]: I

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -174,59 +174,74 @@ Turf and target are seperate in case you want to teleport some distance from a t
...
-		var/length
-		var/pos
-		for(var/datum/objective/objective in O)
```
Comment (eswordthecat) [inline code/__HELPERS/unsorted.dm]: Cri

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -174,59 +174,74 @@ Turf and target are seperate in case you want to teleport some distance from a t
...
-			if(objective.target != mind) continue
```
Comment (eswordthecat) [inline code/__HELPERS/unsorted.dm]: Errytime.

File: code/__HELPERS/unsorted.dm (hunk 2 above)
Comment (ComicIronic) [inline code/__HELPERS/unsorted.dm]: That code doesn't even work.
Paths have no type, so Byond skips them in type loops.

Comment (eswordthecat) [inline code/__HELPERS/unsorted.dm]: What's typesof

**Comment by eswordthecat** [comment]
Can be pulled if success

**Pull request merged**